### ISSUE TITLE
GCC 9.2.1 (Ubuntu 19.20) generates additional warnings and error

### ISSUE THREAD

**Issue by zebity** [collectd/collectd#3341]
*   Version of collectd: 5.9 / 5.10
*   Operating system / distribution: Ubuntu 19.20 (Debian Build)
*   Kernel version (if applicable): Linux new 5.3.0-19-generic

## Expected behavior

NOTE: This is build issue so behaviour relates to building of collectd not it execution.

Successful full and complete build of collectd and its plugins.

## Actual behavior

Build failure due to additional warnings from GCC 9.2.1 Compiler.

Two Cases: DPDK & SNMP Plugins.

  ```
CC       src/dpdkevents_la-dpdkevents.lo
In file included from /usr/include/dpdk/rte_ethdev.h:158,
                 from src/dpdkevents.c:45:
/usr/include/dpdk/rte_ether.h: In function ‘is_broadcast_ether_addr’:
/usr/include/dpdk/rte_ether.h:149:2: error: converting a packed ‘const struct ether_addr’ pointer (alignment 1) to a ‘unaligned_uint16_t’ {aka ‘const short unsigned int’} pointer (alignment 2) may result in an unaligned pointer value [-Werror=address-of-packed-member]
  149 |  const unaligned_uint16_t *ea_words = (const unaligned_uint16_t *)ea;
      |  ^~~~~
cc1: all warnings being treated as errors
```

```
  CC       src/snmp_la-snmp.lo
  CCLD     snmp.la
  CC       src/snmp_agent_la-snmp_agent.lo
In file included from /usr/include/string.h:494,
                 from ./src/daemon/collectd.h:51,
                 from src/snmp_agent.c:30:
In function ‘strncat’,
    inlined from ‘snmp_agent_build_name’ at src/snmp_agent.c:965:9,
    inlined from ‘snmp_agent_format_name’ at src/snmp_agent.c:1030:13,
    inlined from ‘snmp_agent_form_reply.isra.0’ at src/snmp_agent.c:1092:9:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:136:10: error: ‘__builtin___strncat_chk’ output may be truncated copying between 0 and 127 bytes from a string of length 127 [-Werror=stringop-truncation]
  136 |   return __builtin___strncat_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

(Description of the behavior / output that you observed)

## Steps to reproduce

*   step 1 - Install Ubuntu 19.10
*   step 2 - Install collectd build tools and plugin library dependencies
*   step 3 - clone repository
*   step 4 - cd collectd, ./build.sh, ./configure, make
*   step 5 - ./build.sh, ./configure, make

I am raising this issue, as it is an example where building with warnings -> errors is cause build failures due to code that is outside of collectd (in both cases the problem stems from inline functions with included headers from external projects

So these issues cannot be addressed by the collectd team. In the case of the DPDK case the code had not be touch for 4 years and yet with GCC 9.2.1 this is now causing build failures.

My work-around in this case was just to remove the modules from my scope of build: libsnmp-dev & libdpdk-dev

Purpose of logging issue is to provide another example of impact of treating warnings as errors.

NOTE: got to this point as I have completed fixes for other sysctl cases, now need to verify on FreeBSD (build) and validate on Ubuntu 19.10 execute.

**Comment by bzed** [comment]
@zebity the proper thing to do would be to fix these erros instead of disabling the error trigger ;-)

**Comment by zebity** [comment]
@bzed yes but the error are coming from external packages (dpdk & simp) and well outside the reach of collectd which is just consuming the interfaces they expose.

This is one of the "problems" of collectd maintenance, it success means that is it has dependencies to a very large number of external code.

I do not believe that it makes sense to touch this stuff as part of collectd, as it is getting into areas of specialisation that are beyond remit of collectd maintenance.

Though use I do agree ultimately everything should be "fixed", the question is by whom ;-)

**Comment by mrunge** [comment]
In my understanding, these issues are in collectds source code and thus can/should be fixed here.

**Comment by rubenk** [comment]
They are. I've said this before to @zebity, send patches instead of opening issues for every warning if you want to help.

**Comment by zebity** [comment]
@rubenk sometimes I believe it is best to "let sleeping dogs lie". But it seems that the simple matter of raising issues as you come across them is now deemed "incorrect behaviour".

I am only a very minor contributor here and contributing because I have a specific need and have been doing my small set of updates to try to get these satisfied. In doing this in conjunction with others there has been some useful advancement in the overall project. To my mind this is how things should work. Some people do big things others small things.

To @rubenk 's point of sending patches, yes I have and these have been driven by issues mostly.

As a beginner on collectd I try to clearly document the why and what I have done, as this is helpful for people like @mrunge @rpv-tomsk who do a lot of code reviews. Apologies if my style its annoying noisy .

So on the issue of this particular item, here it the code that is generating the warning:
http://git.dpdk.org/dpdk-stable/tree/lib/librte_net/rte_ether.h

```
static inline int is_unicast_ether_addr(const struct ether_addr *ea)
{
	return ((ea->addr_bytes[0] & ETHER_GROUP_ADDR) == 0);
}

```

This code is from Intel dpdk project, so why are we suddenly getting warnings ??
I believe this is a result of move from GCC 8.X to 9.X .

Given code is outside of collectd and I have no access to the CI build tools / settings, I raised issue, as others might come across this soon as get greater adoption of 9.X.

Where I could do distribution workaround (in pkg-debian) I added test for 9.X and added no-warning exceptions.

My intent in raising issues is not to get others to fix them but to provide a trail of useful information to others.

This comment is way, way to long and kind of unnecessary.
People are contributing as they can.

**Comment by rubenk** [comment]
If you believe the warnings come from code outside the Collectd project, why don't your report them there?

You seem to believe these warnings are actual issues, but they're not. I appreciate you trying to help, but what you're doing is not helping. We don't need a `trail of useful information`, we need bugfixes for actual issues and help reviewing open PRs.
